src/hotspot/cpu/aarch64/register_aarch64.hpp

```diff
@@ -32,43 +32,64 @@
 class VMRegImpl;
 typedef VMRegImpl* VMReg;
 
+// Macros to help define all kinds of registers
+
+#define REGISTER_IMPL_DECLARATION(type, name)                           \
+inline const type as_ ## type(int encoding) {                           \
+  assert(encoding <= name::number_of_declared_registers, "invalid register"); \
```

Comment:
Should this be `<` instead of `<=`?

Reply:
No, because `noreg` is a pointer one past the end of the array,